controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -125,32 +129,49 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
 }
 
 func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
-
+	r.storageConsumer.Status.CephResources = []*ocsv1alpha1.CephResourcesSpec{}
 	if !r.storageConsumer.Spec.Enable {
-		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDisabled
-		return reconcile.Result{}, nil
+		if err := r.reconcileNonEnabledPhases(); err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		if err := r.reconcileEnabledPhases(); err != nil {
+			return reconcile.Result{}, err
+		}
 	}
+	return reconcile.Result{}, nil
+}
 
-	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
-	r.storageConsumer.Status.CephResources = []*ocsv1alpha1.CephResourcesSpec{}
-
+func (r *StorageConsumerReconciler) reconcileNonEnabledPhases() error {
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDisabled
```

Comment:
We need to change "Disabled" to something else. The term disabled does not fit this use case anymore

Reply:
done.